### PR TITLE
tests: generic detection of gadget and kernel snaps

### PR DESCRIPTION
**Pull request by plars**
Fix detection of gadget/kernel snap by using the line coming from 'snap list'. This also fixes cases where the kernel name does not match the ${gadget}-kernel pattern.

### CODE REVIEW

**Comment by plars** [comment]
good point, if you want, I can update it to just grep for 'kernel$' an 'gadget$', that way it only catches it at the end of the line.

**Comment by plars** [comment]
I pushed a fix that had it only detect if kernel/gadget is at the end of the line, but this shouldn't have fixed anything with the test failure that we saw before. It's not clear to me why that failed, but it doesn't seem to be failing now, so something in the test or the travis-ci environment is flaky perhaps?

File: tests/lib/names.sh [hunk 1]
```diff
@@ -1,8 +1,4 @@
 #!/bin/bash
 
-gadget_name=$(snap list | sed -n 's/^\(pc\|pi[23]\|dragonboard\|cm3\) .*/\1/p')
-kernel_name=$gadget_name-kernel
-
-if [ "$kernel_name" = "pi3-kernel" ] || [ "$kernel_name" = "cm3-kernel" ]; then
-    kernel_name=pi2-kernel
-fi
+gadget_name=$(snap list | grep 'gadget$' | awk '{ print $1 }')
```
Comment (zyga) [inline tests/lib/names.sh]: Nitpick, we could look for `-gadget$` and `-kernel$` respectively.

Comment (plars) [reply tests/lib/names.sh]: I don't think we want that, there's no '-' in the 'Notes' column that we are matching here. Here's an example of what's matched:
pc               16.04-0.8      9     canonical            gadget
pc-kernel        4.4.0-112.135  102   canonical            kernel

**Pull request merged**